### PR TITLE
drivers: i2c: Microchip MEC172x support device tree clock frequency

### PR DESCRIPTION
**Pull request by jvasanth1**
Modify the Microchip MEC172x I2C driver to use the device tree I2C
clock-frequency property and driver initialization time. Also, fixed
missing idle scaling register programming.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

File: drivers/i2c/i2c_mchp_xec_v2.c [hunk 1]
```diff
@@ -77,11 +79,13 @@ struct xec_speed_cfg {
 	uint32_t start_hold_time;
 	uint32_t config;
 	uint32_t timeout_scale;
+	uint32_t idle_scale;
```
Comment (albertofloyd) [inline drivers/i2c/i2c_mchp_xec_v2.c:82]: Where is this new parameter used?

File: drivers/i2c/i2c_mchp_xec_v2.c [hunk 2]
```diff
@@ -263,6 +267,7 @@ static int i2c_xec_reset_config(const struct device *dev)
 	regs->RSHTM = xec_cfg_params[data->speed_id].start_hold_time;
 	regs->DATATM = xec_cfg_params[data->speed_id].data_timing;
 	regs->TMOUTSC = xec_cfg_params[data->speed_id].timeout_scale;
+	regs->IDLSC = xec_cfg_params[data->speed_id].config;
```
Comment (albertofloyd) [inline drivers/i2c/i2c_mchp_xec_v2.c:270]: This seems unrelated to the clock frequency change. Though it seems correct, seems it has been missing for a while. 
Is this a fix for something else?

Comment (jvasanth1) [reply drivers/i2c/i2c_mchp_xec_v2.c:270]: Yes, this change is to program the idle scaling register. It was missing so far.

File: drivers/i2c/i2c_mchp_xec_v2.c (hunk 1 above)
Comment (jvasanth1) [reply drivers/i2c/i2c_mchp_xec_v2.c:82]: This parameter is used to program the idle scaling register. There was already the `config `field used for that. I will remove that field and use `idle_scale `instead to make it more clear.

Comment (jvasanth1) [reply drivers/i2c/i2c_mchp_xec_v2.c:80]: updated.

**Review by teburd — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
